pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,228 @@
+        final URI controllerURI = URI.create(uriString);
+
+        final String truststorePath =
+                cmd.getOptionValue("truststore") == null ? Constants.DEFAULT_TRUSTSTORE_PATH : cmd.getOptionValue("truststore");
+        final boolean validateHostname = cmd.getOptionValue("validatehost") == null ? false : true;
+
+        final String username = cmd.getOptionValue("username") == null ? Constants.DEFAULT_USERNAME : cmd.getOptionValue("username");
+        final String password = cmd.getOptionValue("password") == null ? Constants.DEFAULT_PASSWORD : cmd.getOptionValue("password");
+
+        SecureBatchReader reader = new SecureBatchReader(scope, stream, controllerURI,
+                truststorePath, validateHostname,
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    public void run() {
+        try (BatchClientFactory batchClient = BatchClientFactory.withScope(scope, clientConfig)) {
+            System.out.println("Done creating batchClient for " + scope + "/" + stream);
+            ArrayList<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(Stream.of(scope, stream), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+            readFromRanges(batchClient, ranges);
+            System.out.println("Done reading ranges of size: " + String.valueOf(ranges.size()));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private int readFromRanges(BatchClientFactory batchClient, List<SegmentRange> ranges) {
+        List<CompletableFuture<Integer>> eventCounts = new ArrayList<>();
+        JavaSerializer<String> serializer = new JavaSerializer<String>();
```

Comment:
Please simplify this code for the benefit of users. See a simpler example of batch reader here: 

https://github.com/pravega/pravega/blob/6b98ef07cbaeb997c930815f32002d59a76186ff/test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java#L219